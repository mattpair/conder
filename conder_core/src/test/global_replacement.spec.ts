import { make_replacer, MONGO_REPLACER } from "../../index"


describe("mongo", () => {
    const replace = make_replacer(MONGO_REPLACER)

    it("get field with mongo specific op", () => {
        expect(replace({
            kind: "GetField",
            target: {kind: "GlobalObject", name: "global"},
            field_name: [{kind: "String", value: "field"}]
        })).toMatchSnapshot()
    })
})