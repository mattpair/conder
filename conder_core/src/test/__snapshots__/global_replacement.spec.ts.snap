// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mongo get field with mongo specific op 1`] = `
Object {
  "key": Array [
    Object {
      "kind": "String",
      "value": "field",
    },
  ],
  "kind": "GetKeyFromObject",
  "obj": "global",
}
`;
